app/services/submission/trait_score_parser.rb

```diff
@@ -71,6 +75,10 @@ def parse_scores
     @upload.log "Parsed #{score_count} scores for #{plant_count} plants, in total."
   end
 
+  def csv
+    @csv || (@csv = CSV.new(input))
```

Comment:
Don't we usually write it like this: `@csv ||= CSV.new(input)`? Nothing wrong with it, just surprised me a bit.